docs/Parameters.rst

```diff
@@ -538,6 +538,18 @@ Learning Control Parameters
 
       -  note that the parent output ``w_p`` itself has smoothing applied, unless it is the root node, so that the smoothing effect accumulates with the tree depth
 
+-  ``interaction_constraints`` :raw-html:`<a id="interaction_constraints" title="Permalink to this parameter" href="#interaction_constraints">&#x1F517;&#xFE0E;</a>`, default = ``""``, type = string
+
+   -  controls which features can appear in the same branch
+
+   -  for CLI, specify as lists separated by commas, e.g. ``[0,1,2],[2,3]``
+
+   -  for Python, specify as list of lists, e.g. ``[[0, 1, 2], [2, 3]]``
+
+   -  for R, specify as a list of character or numeric vectors, e.g. ``list(c("var1", "var2", "var3"), c("var3", "var4"))`` or ``list(c(1L, 2L, 3L), c(3L, 4L))``
```

Comment:
Can you clarify here that the numeric vectors need to be 0-based? R vectors are 1-based so I could see that leading to confusion.

Something like this (but feel free to change the language)

> If numeric, should be 0-based. So `0` means "the first feature", `1` means "the second feature", etc.

Reply:
Actually it expects 1-based vectors then converts to zero-based, similar to set_categorical_feature (where I copied most of the code from). I have updated the docs to clarify that.

Reply:
oh great, ok thanks!